### PR TITLE
Issue #948: [UX] Improve/clean up the 'Add x' links in Admin Bar.

### PR DESCRIPTION
**Pull request by hellomrcat**
PR for https://github.com/backdrop/backdrop-issues/issues/948

### CODE REVIEW

File: core/modules/menu/menu.module [hunk 1]
```diff
@@ -81,14 +81,24 @@ function menu_menu() {
...
+    );
+  }
+  // $items['admin/structure/menu/manage/%menu/add'] = array(
+  //   'title' => 'Add link',
+  //   'page callback' => 'backdrop_get_form',
+  //   'page arguments' => array('menu_edit_item', 'add', NULL, 4),
+  //   'access arguments' => array('administer menu'),
+  //   'type' => MENU_LOCAL_ACTION,
+  //   'file' => 'menu.admin.inc',
+  // );
```
Comment (klonos) [inline core/modules/menu/menu.module]: This corrects the search results so the menu entries are now _Structure: [menu_name]: Add link_, but it also changes the actual menu items too. So from:

_Structure_ -> _Menus_ -> _[menu_name]_ -> _Add link_

...you now get:

_Structure_ -> _Menus_ -> _[menu_name]: Add link_

Notice how the "Add link" menu items have now moved one level up. Not good. I though that the idea here was to add the `[menu_name]:` part **only** to the search results while leaving the menu items as they are.

File: core/modules/node/node.module [hunk 2]
```diff
@@ -1664,7 +1664,7 @@ function node_menu() {
   foreach (node_type_get_types() as $type) {
     $type_url_str = str_replace('_', '-', $type->type);
     $items['node/add/' . $type_url_str] = array(
-      'title' => $type->name,
+      'title' => 'Add content: ' . $type->name,
```
Comment (klonos) [inline core/modules/node/node.module]: ...same as the menu items case. This also changes the actual menu items from:

_Content_ -> _Add content_ -> _[content_type_name]_

to:

_Content_ -> _Add content_ -> _Add content: [content_type_name]_

Not good. We need to change **only** the search results by inserting the "Add content:" part between the "Content: " and "[content_type_name]" parts.

**Comment by hellomrcat** [comment]
@klonos my bad, I'll try to just change the search results.

**Comment by klonos** [comment]
Thanx. Let me know so I can test.

**Comment by jenlampton** [comment]
New PR at https://github.com/backdrop/backdrop/pull/1162

**Pull request closed without merging**